General/Operations/speed.py

```diff
@@ -1,15 +1,54 @@
 # Provides functionality to change the speed
 
-# Created by: Bryce Ilcan, november 7 2024
+# Created by: Bryce Ilcan, November 7th, 2024
+# Valid Speed Check added by: Nathan Jourdain, November 30th, 2024
 
 #TODO: Conduct safety checks with physical vehicle [IMPORTANT!]
 
 from pymavlink import mavutil
+import General.Operations.initialize as initialize
+
+vehicle_connection, valid_connection= initialize.connect_to_vehicle('udpin:127.0.0.1:14550')
+
+max_speed = 55 / 3.6 # 55 km/h converted to m/s
+min_speed = 40 / 3.6 # 40 km/h converted to m/s
+
+vehicle_connection.mav.param_set_send(
+        vehicle_connection.target_system,
+        vehicle_connection.target_component,
+        b'AIRSPEED_MAX',
+        max_speed,
+        mavutil.mavlink.MAV_PARAM_TYPE_UINT32
+    )
+
+vehicle_connection.mav.param_set_send(
+        vehicle_connection.target_system,
+        vehicle_connection.target_component,
+        b'AIRSPEED_MIN',
+        min_speed,
+        mavutil.mavlink.MAV_PARAM_TYPE_UINT32
+    )
+
+def check_valid_speed(speed):
+    #PROMISES: Returns 0 if the speed value is not between the max and min values
+    #REQUIRES: "speed" is a positive, real number
+    
+    if(speed > max_speed):
+        print("ERROR: requested speed above maximum allowed speed")
+        return 0
```

Comment:
Typically, if an error occurs, we want to return an exit code of 1. If it is successful, that is when we return an exit code of zero. So if we could reverse the 0 and 1 return codes, that'd be great.